components/brave_extension/extension/brave_extension/content.ts

```diff
@@ -1,24 +1,93 @@
 const unique = require('unique-selector').default
 
 let target: EventTarget | null
+let genericExceptions: Array<string> | undefined = undefined
+let classIdBuffer: { classes: string[], ids: string[] } = { classes: [], ids: [] }
 
 function getCurrentURL () {
   return window.location.hostname
 }
 
-/*function applyCosmeticFilterMutationObserver (filterList: any) {
+const getClassesAndIds = (function () {
```

Comment:
Is there a reason why this is an Immediately Invoked Function Expression that returns another function? I worry that we'll lose out on some v8 optimizations when this is performance sensitive, and I don't see a reason this can't be a straightforward function with the top-level variables global (to this content script).

Reply:
Just for encapsulation - if it works better with global variables that's fine with me